src/Mod/Path/PathTests/TestPathSlot.py

```diff
@@ -0,0 +1,264 @@
+# *                                                                         *
+# *   This program is distributed in the hope that it will be useful,       *
+# *   but WITHOUT ANY WARRANTY; without even the implied warranty of        *
+# *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         *
+# *   GNU Library General Public License for more details.                  *
+# *                                                                         *
+# *   You should have received a copy of the GNU Library General Public     *
+# *   License along with this program; if not, write to the Free Software   *
+# *   Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  *
+# *   USA                                                                   *
+# *                                                                         *
+# ***************************************************************************
+
+import FreeCAD
+import Path
+import PathScripts.PathJob as PathJob
+import PathScripts.PathSlot as PathSlot
+from PathTests.PathTestUtils import PathTestBase
+
+
+class TestPathSlot(PathTestBase):
+    """Unit tests for the Slot operation."""
+
+    @staticmethod
+    def _setup_test_environment():
+        """_setup_test_environment()...
+        This method is upon instantiation of this test class.  Add code and objects here
+        that are needed for the duration of the test() methods in this class.
+        This method does not have access to the class `self` reference.
+        """
+        # doc = FreeCAD.open(FreeCAD.getHomePath() + 'Mod/Path/PathTests/test_slot_01.fcstd')
+        doc_title = "TestSlot"
+        doc = FreeCAD.newDocument(doc_title)
+
+        # This is a real world test to make sure none of the tool chain broke
+        box0 = doc.addObject('Part::Box', 'Box')
+        box0.Length = 25.0
+        box0.Width = 50.0
+        box0.Height = 10.0
+        box1 = doc.addObject('Part::Box', 'Box')
+        box1.Length = 5.0  # X
+        box1.Width = 15.0  # Y
+        box1.Height = 5.0  # Z
+        box1.Placement = FreeCAD.Placement(FreeCAD.Vector(5.0, 0.0, 5.0), FreeCAD.Rotation(FreeCAD.Vector(0,0,1), 0))
+        doc.recompute()
+        cut = doc.addObject('Part::Cut', 'Cut')
+        cut.Base = box0
+        cut.Tool = box1
+        doc.recompute()
+
+        # Create Job object
+        PathJob.Create('Job', [doc.Cut], None)
+        doc.recompute()
+
+    @classmethod
+    def setUpClass(cls):
+        """setUpClass()...
+        This method is called upon instantiation of this test class.  Add code and objects here
+        that are needed for the duration of the test() methods in this class.
+        This method does not have access to the class `self` reference.  This method
+        is able to call static methods within this same class.
+        """
+        # FreeCAD.Console.PrintMessage("TestPathSlot.setUpClass()\n")
+        cls._setup_test_environment()
```

Comment:
why this layer of indirection?

Reply:
I will remove this and combine the methods and add additional comments to explain the purpose of the code.  The one-liner seen here is a nice clear indication of what is being done - that was my thinking, as well as educating those who follow after and want to learn from this unit test.